api/v1/inferencepool_types.go

```diff
@@ -246,21 +255,22 @@ type ParentGatewayReference struct {
 	//
 	// +optional
 	// +kubebuilder:default="gateway.networking.k8s.io"
-	Group *Group `json:"group"`
+	Group *Group `json:"group,omitempty"`
 
 	// Kind is kind of the referent. For example "Gateway".
 	//
 	// +optional
 	// +kubebuilder:default=Gateway
-	Kind *Kind `json:"kind"`
+	Kind Kind `json:"kind,omitempty"`
```

Comment:
Kind and Group are two fields in the same struct with the same behavior, but one is a pointer and one is not... why?